cmd/az-analyze/cmd/azva.go

```diff
@@ -0,0 +1,244 @@
+	Use:   "azva",
+	Short: "Azure Volume Attachment",
+	Long:  `Azure Volume Attachment is a Kubernetes Custom Resource.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// typesFlag := []string{"pod", "node", "zone", "namespace"}
+		// valuesFlag := []string{pod, node, zone, namespace}
+
+		// for _, value := range valuesFlag {
+
+		// }
+
+		pod, _ := cmd.Flags().GetString("pod")
+		node, _ := cmd.Flags().GetString("node")
+		zone, _ := cmd.Flags().GetString("zone")
+		namespace, _ := cmd.Flags().GetString("namespace")
+
+		numFlag := cmd.Flags().NFlag()
+		if hasNamespace := namespace != ""; hasNamespace {
+			numFlag--
+		}
+
+		var azva []AzvaResource
+
+		if numFlag > 1 {
+			fmt.Printf("only one of the flags is allowed.\n" + "Run 'az-analyze --help' for usage.\n")
+		} else {
+			if numFlag == 0 {
+				// TODO: the same as  kubectl get AzVolumeAttachment
+				fmt.Println("no flags")
+			} else if pod != "" {
+				azva = GetAzVolumeAttachementsByPod(pod, namespace)
+				displayAzva(azva, "POD")
+			} else if node != "" {
+				azva = GetAzVolumeAttachementsByNode(node)
+				displayAzva(azva, "NODE")
+			} else if zone != "" {
+				azva = GetAzVolumeAttachementsByZone(zone)
+				displayAzva(azva, "ZONE")
+			} else {
+				fmt.Printf("invalid flag name\n" + "Run 'az-analyze --help' for usage.\n")
+			}
+		}
+	},
+}
+
+func init() {
+	getCmd.AddCommand(azvaCmd)
+	azvaCmd.PersistentFlags().StringP("pod", "p", "", "insert-pod-name (only one of the flags is allowed).")
+	azvaCmd.PersistentFlags().StringP("node", "d", "", "insert-node-name (only one of the flags is allowed).")
+	azvaCmd.PersistentFlags().StringP("zone", "z", "", "insert-zone-name (only one of the flags is allowed).")
+	azvaCmd.PersistentFlags().StringP("namespace", "n", "", "insert-namespace (optional).")
+
+	// Here you will define your flags and configuration settings.
+
+	// Cobra supports Persistent Flags which will work for this command
+	// and all subcommands, e.g.:
+	// azvaCmd.PersistentFlags().String("foo", "", "A help for foo")
+
+	// Cobra supports local flags which will only run when this command
+	// is called directly, e.g.:
+	// azvaCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+}
+
+type AzvaResource struct {
+	ResourceType string
+	Namespace    string
+	Name         string
+	Age          time.Duration
+	RequestRole  v1beta1.Role
+	Role         v1beta1.Role
+	State        v1beta1.AzVolumeAttachmentAttachmentState
+}
+
+func GetAzVolumeAttachementsByPod(podName string, namespace string) []AzvaResource {
+	result := make([]AzvaResource, 0)
+
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	// access to config and Clientsets
+	config := getConfig()
+	clientsetK8s := getKubernetesClientset(config)
+	clientsetAzDisk := getAzDiskClientset(config)
+
+	// get pvc claim names of pod
+	pvcClaimNameSet := make(map[string]bool)
+
+	pods, err := clientsetK8s.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{})
```

Comment:
Similar comment as above, it would be more efficient to get only one pod when pod name is given  and list all the pods if no podName is provided.

Reply:
Edited to get only one pod when pod name is given, but `get azva --pod` is not supported since flag value must be given.
Added new feature `get azva` to list the information for all pods/nodes/zones.